src/test/java/ru/itmo/sd/deadliner2bot/service/TaskFetchingServiceJpaTest.java

```diff
@@ -0,0 +1,178 @@
+        Set<TodoNotification> todoNotifications = taskFetchingService
+                .getTodoNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(todoNotifications).contains(todoNotification);
+    }
+
+    @Test
+    @DisplayName(value = "TODO notifications out of range not fetched")
+    public void todoNotificationNotInRangeNotFetched() {
+        Chat chat = createChat();
+        createTodoWithNotification(chat, now.plusSeconds(2L * storedRangeSec));
+
+        Set<DailyNotification> dailyNotifications = taskFetchingService
+                .getDailyNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(dailyNotifications).isEmpty();
+        Set<TodoNotification> todoNotifications = taskFetchingService
+                .getTodoNotifications(now.plusSeconds(storedRangeSec));
+        assertThat(todoNotifications).isEmpty();
+    }
+
+    @Test
+    @DisplayName(value = "Daily notification added in scheduled thread pool")
+    public void dailyNotificationAddedInScheduledThreadPool() {
+        Chat chat = createChat();
+        LocalDateTime notificationTime = now.plusSeconds(storedRangeSec / 2);
+        createDailyNotification(chat, notificationTime);
+
+        taskFetchingService.fetchNotifications();
+        ArgumentCaptor<TaskFetchingService.DailyNotificationRunnable> argumentCaptor =
+                ArgumentCaptor.forClass(TaskFetchingService.DailyNotificationRunnable.class);
+        verify(threadPoolTaskScheduler, times(1))
+                .schedule(argumentCaptor.capture(),
+                        eq(Timestamp.valueOf(notificationTime)));
+        TaskFetchingService.DailyNotificationRunnable runnable = argumentCaptor.getValue();
+        runnable.run();
+        verify(bot, times(1)).sendMessage(eq(chat.getChatId()), any());
+    }
+
+    @Test
+    @DisplayName(value = "TODO notification added in scheduled thread pool")
+    public void todoNotificationAddedInScheduledThreadPool() {
+        Chat chat = createChat();
+        LocalDateTime notificationTime = now.plusSeconds(storedRangeSec / 2);
+        createTodoWithNotification(chat, notificationTime);
+
+        taskFetchingService.fetchNotifications();
+        ArgumentCaptor<TaskFetchingService.TodoNotificationRunnable> argumentCaptor =
+                ArgumentCaptor.forClass(TaskFetchingService.TodoNotificationRunnable.class);
+        verify(threadPoolTaskScheduler, times(1))
+                .schedule(argumentCaptor.capture(),
+                        eq(Timestamp.valueOf(notificationTime)));
+        TaskFetchingService.TodoNotificationRunnable runnable = argumentCaptor.getValue();
+        runnable.run();
+        verify(bot, times(1)).sendMessage(eq(chat.getChatId()), any());
+    }
+
+    private Chat createChat() {
+        Chat chat = new Chat();
+        chat.setChatId(1);
+        return chatRepository.save(chat);
+    }
+
+    private DailyNotification createDailyNotification(Chat chat, LocalDateTime notificationTime) {
```

Comment:
Looks like almost the same (with slightly different signature) create* functions will be used during bot work (not only for testing). So I suggest to move them to corresponding services in source part.

Reply:
I am highly unsure about this. I suggest refactoring as soon as required.